tools/ea/Ea/Genesis.hs

```diff
@@ -49,87 +53,51 @@ module Ea.Genesis
 import Control.Lens
 
 import Data.Text
+import Data.Word
 
 import Chainweb.Graph
 import Chainweb.Test.TestVersions
 import Chainweb.Version
 import Chainweb.Version.Development
+import Chainweb.Version.FastDevelopment
 import Chainweb.Version.Mainnet
 import Chainweb.Version.Testnet
 
 
 -- ---------------------------------------------------------------------- --
 -- Genesis Tx Data
 
--- | Type-safe representation of chain ids
--- for a 10-chain graph.
---
-data GChainId
-    = Zero
-    | One
-    | Two
-    | Three
-    | Four
-    | Five
-    | Six
-    | Seven
-    | Eight
-    | Nine
-    | N
-    | KAD
-    deriving (Eq, Ord, Enum)
-
-instance Show GChainId where
-  show = \case
-    Zero -> "0"
-    One -> "1"
-    Two -> "2"
-    Three -> "3"
-    Four -> "4"
-    Five -> "5"
-    Six -> "6"
-    Seven -> "7"
-    Eight -> "8"
-    Nine -> "9"
-    N -> "N"
-    KAD -> "KAD"
+-- A range of chain IDs [l,r]
+data ChainIdRange
+    = ChainIdRange !Word32 !Word32
+    deriving (Eq, Ord)
+
+chainIdRangeTag :: ChainIdRange -> String
+chainIdRangeTag (ChainIdRange l u)
+    | l == u = show l
+    | otherwise = show l <> "to" <> show u
```

Comment:
Is e.g. `show (ChainIdRange 5 7) == "5to7"` intentional? Did you mean that it should be `"5 to 7"`? Not sure that it matters.

Reply:
Yeah this is intentional, it shows up in the files that contain the genesis payloads. `Mainnet10to19.hs` for example.